Firebase/Messaging/FIRMessagingPendingTopicsList.m

```diff
@@ -109,20 +109,22 @@ + (void)pruneTopicBatches:(NSMutableArray <FIRMessagingTopicBatch *> *)topicBatc
   }
 }
 
-#pragma mark NSCoding
+#pragma mark NSSecureCoding
+
++ (BOOL)supportsSecureCoding {
+  return YES;
+}
 
 - (void)encodeWithCoder:(NSCoder *)aCoder {
   [aCoder encodeObject:[NSDate date] forKey:kPendingTopicsTimestampEncodingKey];
   [aCoder encodeObject:self.topicBatches forKey:kPendingBatchesEncodingKey];
 }
 
 - (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
-
   if (self = [self init]) {
-    _archiveDate = [aDecoder decodeObjectForKey:kPendingTopicsTimestampEncodingKey];
-    NSArray *archivedBatches = [aDecoder decodeObjectForKey:kPendingBatchesEncodingKey];
-    if (archivedBatches) {
-      _topicBatches = [archivedBatches mutableCopy];
+    _archiveDate = [aDecoder decodeObjectOfClass:[NSDate class] forKey:kPendingTopicsTimestampEncodingKey];
+    _topicBatches = [aDecoder decodeObjectOfClass:[NSMutableArray<FIRMessagingTopicBatch *> class] forKey:kPendingBatchesEncodingKey];
```

Comment:
The same here, we should pass `FIRMessagingTopicBatch` class like [here](https://github.com/firebase/firebase-ios-sdk/blob/f138a75a33cde26a58e586274e2dc7f8618fd924/GoogleDataTransport/GDTCORLibrary/GDTCORStorage.m#L314)

Reply:
you mean instead of NSMutableArray<FIRMessagingTopicBatch *>, use [NSSet setWithObjects:[NSMutableArray class], [FIRMessagingTopicBatch class], nil];

Reply:
yep.